### PR TITLE
added plugin development security doc

### LINKED ISSUE #185: Document the security best practice
We need to document the security best practice to tell users how to use Notation in a security posture. The security best practice and considerations include but are not limited to:

- Security considerations for using Notation plugins
- Security considerations for using container registries
- Notation authentication (Credential Store)
- What should users do against the threat model https://github.com/notaryproject/notaryproject/pull/242

This is a big topic. We can iterate on the content creation.

### LINKED ISSUE #227: Document security best practices for plugin development
As part of the release of Notation 1.0.0 we need to document the security best practices to develop plugins for Notation. We should explain more into details how the communication between the Notation CLI and the plugin is handled and how plugin developers can verify that they communicate with a legit Notation installation. Also, we should document the best practices to publish plugins to customers can verify the authenticity o9f the plugin (related to #225).

### PR DESCRIPTION
**Pull request by zr-msft**
fixes #227 
fixes #185

### CODE REVIEW

**Comment by zr-msft** [comment]
@yizha1  @FeynmanZhou @shizhMSFT 

This is a very early version of the doc. Can you please provide additional suggestions with details for anything else that should be included.

**Comment by shizhMSFT** [comment]
Most best practices are actually covered by the plugin spec itself. It would be better to let the plugin developers to read the specs carefully.

File: content/en/docs/concepts/secure-plugin-development.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+plugins run as a binary on the system, invoked by notation. This means that the plugin has access to the same resources as the user running notation. This includes access to the filesystem, network, and other resources. Ensure you are only accessing resources that are required for your plugin to function.
+
+In some cases, your plugin may have access to sensitive data such as private keys. Ensure that you are not leaking this data to the filesystem, network, or other resources, including `stdout` and `stderr`.
+
+If using the `pluginConfig` key/value store, do not store sensitive data in as the data is not encrypted and is stored as plaintext.
+
+
+## Securely distributing plugins
+
```
Comment (FeynmanZhou) [inline content/en/docs/concepts/secure-plugin-development.md:18]: Suggest adding checksum verification as a security best practice. For example, the Notation Azure Key Vault plugin has a checksum file in its releases: https://github.com/Azure/notation-azure-kv/releases

File: content/en/docs/concepts/secure-plugin-development.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+## General security guidance
+
+plugins run as a binary on the system, invoked by notation. This means that the plugin has access to the same resources as the user running notation. This includes access to the filesystem, network, and other resources. Ensure you are only accessing resources that are required for your plugin to function.
+
+In some cases, your plugin may have access to sensitive data such as private keys. Ensure that you are not leaking this data to the filesystem, network, or other resources, including `stdout` and `stderr`.
+
+If using the `pluginConfig` key/value store, do not store sensitive data in as the data is not encrypted and is stored as plaintext.
+
```
Comment (FeynmanZhou) [inline content/en/docs/concepts/secure-plugin-development.md:15]: Suggest reminding developers to follow [plugin development Spec](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) strictly when developing a plugin.

File: content/en/docs/concepts/secure-plugin-development.md (hunk 1 above)
Comment (zr-msft) [reply content/en/docs/concepts/secure-plugin-development.md:25]: fixed

File: content/en/docs/concepts/secure-plugin-development.md (hunk 2 above)
Comment (zr-msft) [reply content/en/docs/concepts/secure-plugin-development.md:23]: fixed

**Comment by zr-msft** [comment]
@shizhMSFT @FeynmanZhou thank you for the quick feedback!

I've incorporated your feedback into the PR.

PTAL

File: content/en/docs/concepts/secure-plugin-development.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+The following guidance is not a substitute for the Notary Project specification for plugins. It is intended to highlight key security considerations for developing and distributing plugins for Notation.
+{{% /alert %}}
+
+## General security guidance
+
+In general when developing your plugin, ensure you strictly adhere to the [Notary Project specification for plugins](https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md).
+
+Plugins run as a binary and are invoked by Notation. This means that the plugin has access to the same resources as the user running Notation. This includes access to the filesystem, network, and other resources. Ensure you are only accessing resources that are required for your plugin to function.
+
+In some cases, your plugin may have access to sensitive data such as private keys. Ensure that you are not leaking this data to the filesystem, network, or other resources, including `stdout` and `stderr`.
```
Comment (yizha1) [inline content/en/docs/concepts/secure-plugin-development.md:20]: @shizhMSFT  @priteshbandi  could you review the following text?
> Plugins run as a binary and are invoked by Notation. This means that the plugin has access to the same resources as the user running Notation. This includes access to the filesystem, network, and other resources. Ensure you are only accessing resources that are required for your plugin to function.
> 
> In some cases, your plugin may have access to sensitive data such as private keys. Ensure that you are not leaking this data to the filesystem, network, or other resources, including `stdout` and `stderr`.

**Review by yizha1 — COMMENTED** [review]
Thanks. @zr-msft LTGM for first version. Just need Shiwei and Pritesh to confirm the text I quoted.

File: content/en/docs/concepts/secure-plugin-development.md (hunk 3 above)
Comment (priteshbandi) [reply content/en/docs/concepts/secure-plugin-development.md:20]: LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

File: content/en/docs/concepts/secure-plugin-development.md (hunk 3 above)
Comment (shizhMSFT) [reply content/en/docs/concepts/secure-plugin-development.md:20]: LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**